db/src/blocks.rs

```diff
@@ -133,11 +133,19 @@ where
     }
 
     /// Returns an iterator of `Id`s sorted by least recently used.
-    pub fn lru(&self) -> impl Iterator<Item = Result<Id>> {
-        self.lru
-            .iter()
-            .values()
-            .map(|v| v.map(Into::into).map_err(Into::into))
+    pub fn lru(&self) -> impl Iterator<Item = Result<(u64, Id)>> {
+        self.lru.iter().map(|res| {
+            res.map(|(atime, id)| (u64::from(&Id::from(atime)), id.into()))
+                .map_err(Into::into)
+        })
+    }
+
+    /// Returns the current atime.
+    pub fn atime(&self) -> u64 {
+        match self.lru.last() {
```

Comment:
Doesn't this traverse the entire iterator? Or is the sled iterator a double ended iterator that will do this efficiently, and the map retains that property?

Reply:
this isn't an iterator, it's a method of the tree

Reply:
but yes, the iterator would be double ended if I used one.

Reply:
ah, must be me having done too much scala in the past. where there is no distinction between fields and methods.